dartagnan/src/main/java/com/dat3m/dartagnan/program/Function.java

```diff
@@ -47,17 +52,25 @@ public Function(String name, FunctionType type, List<String> parameterNames, int
         }
     }
 
+    @Override
+    public Type getType() {
+        return TypeFactory.getInstance().getArchType();
```

Comment:
why do we fallback to the default arch type for the type of a function?

Reply:
This one is actually not the `FunctionType`, but the expression type. If we use a function as an expression, then we always mean the function's address, i.e., you could do `ptr r = myFunc; call r;`. For all addresses we use the arch type for now, but will eventually use a dedicated pointer type.